docs/adr/0009-infrastructure-as-code-sharing-approach.md

```diff
@@ -0,0 +1,94 @@
+# infrastructure-as-code-sharing-approach
+
+## Status
+
+Proposed
+
+## Context
+
+* The current version of Marain uses ARM templates to automate the infrastructure deployment for each Marain service.
+* The deployment templates resides alongside the code for each service
+* There is significant commonality/duplication of ARM templates between services
+* We wish to migrate to using Bicep to define the Marain infrastructure deployments
+* A desire to minimise duplication and utilise shared Bicep modules
+* Ideally, the overall Marain deployment process should be as low-friction as the current "'git clone' & deploy" approach
+
+Given this context and goals, the challenge is how to combine the use of shared Bicep modules (to de-duplicate the infrastructure-as-code) whilst still having a low friction means of deploying the Marain stack.  The absence of a defacto public Bicep module repository that can host the Marain-specific modules means there isn't an obvious way to share these modules without additional work.
+
+***NOTE**: A more general, but related, Bicep reuse [ADR](https://github.com/endjin/Endjin.RecommendedPractices.Bicep/blob/main/docs/adr/0002-sharing-bicep-modules.md) was written before it became clear that the public Bicep registry would not be 'open to all' in the same way that nuget.org or Docker Hub are.*
```

Comment:
Minor editing note: GitHub has interpretted the lone '*' at the end of this line as meaning that you wanted the whole of the rest of the document to be in **bold**!